fms2_io/include/domain_write.inc

```diff
@@ -102,35 +102,37 @@ subroutine domain_write_2d(fileobj, variable_name, vdata, unlim_dim_level, &
                                                               !! will be written
                                                               !! in each dimension.
 
-  integer :: xdim_index
-  integer :: ydim_index
-  type(domain2d), pointer :: io_domain
-  integer :: xpos
-  integer :: ypos
-  integer :: xmax
-  integer :: ymax
-  integer :: isd
-  integer :: isc
-  integer :: xc_size
-  integer :: jsd
-  integer :: jsc
-  integer :: yc_size
+  integer(kind=int32), dimension(:,:), allocatable :: buf_int32
```

Comment:
These integers and reals should probably use the [portable kinds defined in platform.h](https://github.com/NOAA-GFDL/FMS/blob/master/include/fms_platform.h)

Reply:
Instead of platform.h, I think you should use [platform_mod](https://github.com/NOAA-GFDL/FMS/blob/master/platform/platform.F90) to avoid the errors of including platform.h.  The portable kinds are defined there.